utilities/xml_parser.py

```diff
@@ -0,0 +1,170 @@
+            testcaseelem = ts.getElementsByTagName("testcase")
+            for tc in testcaseelem:
+                tcname = tc.getAttribute("name")
+                if tc.getElementsByTagName("failure") or tc.getElementsByTagName("error"):
+                    failed = True
+                    failed_tests.append(tcname)
+                else:
+                    failed = False
+                    passed_tests.append(tcname)
+
+    if failed_tests:
+        failed_tests = transform_and_write_to_file(failed_tests, "failed_tests.conf")
+
+    if passed_tests:
+        passed_tests = transform_and_write_to_file(passed_tests, "passed_tests.conf")
+    print(" or ".join(failed_tests))
+    f_file = open("test_failures", "w+")
+    f_file.write(" or ".join(failed_tests))
+    p_file = open("test_passed", "w+")
+    p_file.write(" or ".join(passed_tests))
+    return passed_tests, failed_tests
+
+
+def getNodeText(nodelist):
+    rc = []
+    for node in nodelist:
+        if node.nodeType == node.TEXT_NODE:
+            rc.append(node.data)
+    return ''.join(rc)
+
+
+def check_if_exists(test_list, test_line):
+    new_test_line = ''.join(sorted(test_line))
+    for t in test_list:
+        t1 = ''.join(sorted(t))
+        if t1 == new_test_line:
+            return True, t
+    return False, ""
+
+
+def transform_and_write_to_file(tests_list, filename):
+    new_test_list = []
+    for test in tests_list:
+        line = filter_fields(test)
+        line = line.rstrip(",")
+        isexisted, _ = check_if_exists(new_test_list, line)
+        if not isexisted:
+            new_test_list.append(line)
+
+    file = open(filename, "w+")
+    for line in new_test_list:
+        file.writelines((line) + "\n")
+    file.close()
+    return new_test_list
+
+
+def filter_fields(testname):
+    testwords = testname.split(",")
+    line = ""
+    for fw in testwords:
+        if not fw.startswith("logs_folder") and not fw.startswith("conf_file") \
+                and not fw.startswith("cluster_name:") \
+                and not fw.startswith("ini:") \
+                and not fw.startswith("case_number:") \
+                and not fw.startswith("num_nodes:") \
+                and not fw.startswith("spec:"):
+            if not "\":" in fw or "query:" in fw:
+                # log.info("Replacing : with ={}".format(fw))
+                line = line + fw.replace(":", "=", 1)
+            else:
+                line = line + fw
+            if fw != testwords[-1]:
+                line = line + ","
+
+    return line
+
+
+if __name__ == "__main__":
+    usage = """usage: xml_parser.py
+       usage: python mobile_server_pool.py
+       --file=xml-file path
+       """
+    parser = OptionParser(usage=usage)
+
+    parser.add_option("--xml-file",
+                      action="store", dest="xml_file", default="http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-sgreplicate2/lastCompletedBuild/artifact/results/results.xml",
```

Comment:
how about not setting default? or default location without job name, such as a regex pattern so each job can fill up with its name? just curious.